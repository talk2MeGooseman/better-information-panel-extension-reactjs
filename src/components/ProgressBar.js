import React from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react';

import { withStyles } from '@material-ui/core/styles';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';

const styles = theme => ({
  root: {
    width: '100%',
  },
  button: {
    marginRight: theme.spacing.unit,
  },
  instructions: {
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit,
  },
});

function getSteps() {
  return ['Add your tabs', 'Enter your text', 'Preview', 'Save'];
}

class ProgressBar extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    tabsStore: PropTypes.object.isRequired,
  };

  render() {
    const { classes } = this.props;
    const steps = getSteps();
    const { activeStep } = this.props.tabsStore;

    return (
      <div className={classes.root}>
        <Stepper activeStep={activeStep}>
          {steps.map((label, index) => {
            const props = {
            };
            const labelProps = {};
            return (
              <Step key={label} {...props}>
                <StepLabel {...labelProps}>{label}</StepLabel>
              </Step>
            );
          })}
        </Stepper>
      </div>
    );
  }
}

export default withStyles(styles)(observer(ProgressBar));
