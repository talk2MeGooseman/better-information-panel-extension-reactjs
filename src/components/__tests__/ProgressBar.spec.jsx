import React from "react";
import { shallow, mount } from "enzyme";

import ProgressBar from "../ProgressBar";

describe('ProgressBar', () => {
  it('matches default state snapshot', () => {
    const wrapper = mount(<ProgressBar tabsStore={{}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set step 2 active', () => {
    let tabsStore = {
      activeStep: 2
    };
    const wrapper = mount(<ProgressBar tabsStore={tabsStore} />);
    expect(wrapper).toMatchSnapshot();
  });
});