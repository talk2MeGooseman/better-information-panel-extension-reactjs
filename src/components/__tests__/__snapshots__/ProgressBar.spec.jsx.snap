// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar matches default state snapshot 1`] = `ReactWrapper {}`;

exports[`ProgressBar should set step 2 active 1`] = `ReactWrapper {}`;
